### PR TITLE
SonarLint bugfixes

### PR DESCRIPTION
**Pull request by csuri66**
Fixmaxxed sonarlint detected problems.

### CODE REVIEW

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -30,18 +30,18 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
```
Comment (ladanyiattila) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:33]: Great job

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -30,18 +30,18 @@ public TorpedoStore(int numberOfTorpedos){
...
     }
 
     boolean success = false;
```
Comment (ladanyiattila) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:36]: Delete the value of the variable

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 3]
```diff
@@ -11,7 +11,7 @@ public class TorpedoStore {
 
   // rate of failing to fire torpedos [0.0, 1.0]
   private double FAILURE_RATE = 0.0; //NOSONAR
-
+  Random generator = new Random();
```
Comment (ladanyiattila) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:14]: Great job

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 4]
```diff
@@ -30,18 +30,18 @@ public TorpedoStore(int numberOfTorpedos){
...
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
+    
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (ladanyiattila) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:44]: Great job

**Comment by csuri66** [comment]
BugFix

**Pull request merged**